open-api/rest-catalog-open-api.yaml

```diff
@@ -1966,6 +2018,18 @@ components:
               "GET /v1/{prefix}/namespaces/{namespace}/tables/{table}",
               "GET /v1/{prefix}/namespaces/{namespace}/views/{view}"
             ]
+        idempotency-key-lifetime:
+          type: string
+          format: duration
+          description: >
+            Client reuse window for an Idempotency-Key (ISO-8601 duration, e.g., PT30M, PT24H).
+            Interpreted as the maximum time from the first submission using a key to the last retry
+            during which a client may reuse that key. Servers SHOULD accept retries for at least this
+            duration and MAY include a grace period to account for delays/clock skew. Clients SHOULD NOT
+            reuse an Idempotency-Key after this window elapses; they SHOULD generate a new key for any
+            subsequent attempt. Presence of this field indicates the server supports Idempotency-Key
```

Comment:
Lets say a client retried n times before, was suppose to retry n+1 th time but before it could retry the window expired, there can 2 cases : 
1. n th try succeeded 
2. n th try fail 

n th try failing and we generating new idempotency key for n + 1 th attempt should be fine, but in case when n th try succeeded and we created a new idempotency key, we treat this is as a new request ? we can then self conflict ? 

would it better to fail with commit state unknown here ?

Reply:
Idempotency only covers duplicates within the window; after it expires, the client should read current state first (e.g., check metadata location / snapshot id) and only submit a new request if the desired change isn’t already present.

Reply:
> they SHOULD generate a new key for any subsequent attempt

I see should we be explicit in saying this ? like `submit a new request` and abandon the current retry loop ?